app/templates/base.html

```diff
@@ -47,6 +47,31 @@ <h2>Yalies</h2>
             {% endblock %}
         </main>
         <footer>
+            <button class="scrollToTopBtn">⇪</button>
+            <script>
+                var target = document.getElementById('navbar');
+                var scrollToTopBtn = document.querySelector(".scrollToTopBtn")
+                var rootElement = document.documentElement
+
+                function callback(entries, observer) {
+                entries.forEach(entry => {
+                    if (!entry.isIntersecting) {
+                    scrollToTopBtn.classList.add("showBtn")
+                    } else {
+                    scrollToTopBtn.classList.remove("showBtn")
+                    }
+                });
+                }
+                function scrollToTop() {
+                rootElement.scrollTo({
+                    top: 0,
+                    behavior: "smooth"
+                })
+                }
+                scrollToTopBtn.addEventListener("click", scrollToTop);
+                let observer = new IntersectionObserver(callback);
```

Comment:
I'm not super sure what's going on here, I think it's cleaner to listen for a scroll event and then hide or show the button based on whether, say, `window.scrollTop > 2 * window.innerHeight`.